fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -93,34 +95,45 @@ private boolean doesSchemaContainAllDimensionFields(Dimension dimension, Schema
      *
      * @throws IllegalArgumentException thrown if JSON object `fields` is not present
      */
-    public Set<DimensionRow> parseAvroFileDimensionRows(Dimension dimension, String avroFilePath)
+    public Stream<DimensionRow> parseAvroFileDimensionRows(Dimension dimension, String avroFilePath)
         throws IllegalArgumentException {
 
         // Creates an AVRO DatumReader object
         DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
 
         // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
-        try (DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader)) {
+        try {
+            DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader);
```

Comment:
What happens if this generates an error? Previously, it would have closed the file, and now it won't. That will effectively leak the file, to be closed later during GC. I'm not sure if we care about waiting that long or not.